be/src/exprs/vectorized/cast_expr.cpp

```diff
@@ -28,45 +28,81 @@
 #include "util/json.h"
 
 namespace starrocks::vectorized {
-
-template <PrimitiveType FromType, PrimitiveType ToType>
+#define THROW_RUNTIME_ERROR_WITH_TYPE(TYPE)              \
+    std::stringstream ss;                                \
+    ss << "not supported type " << type_to_string(TYPE); \
+    throw std::runtime_error(ss.str())
+
+#define THROW_RUNTIME_ERROR_WITH_TYPES_AND_VALUE(FROMTYPE, TOTYPE, VALUE) \
+    std::stringstream ss;                                                 \
+    ss << "cast from " << type_to_string(FROMTYPE) << "(" << VALUE << ")" \
+       << " to " << type_to_string(TOTYPE) << " failed";                  \
+    throw std::runtime_error(ss.str())
+
+template <PrimitiveType FromType, PrimitiveType ToType, bool AllowThrowException = false>
 ColumnPtr cast_fn(ColumnPtr& column);
 
 // All cast implements
-#define SELF_CAST(FROM_TYPE)                                      \
-    template <>                                                   \
-    ColumnPtr cast_fn<FROM_TYPE, FROM_TYPE>(ColumnPtr & column) { \
-        return column->clone();                                   \
+#define SELF_CAST(FROM_TYPE)                                             \
+    template <>                                                          \
+    ColumnPtr cast_fn<FROM_TYPE, FROM_TYPE, false>(ColumnPtr & column) { \
+        return column->clone();                                          \
+    }                                                                    \
+    template <>                                                          \
+    ColumnPtr cast_fn<FROM_TYPE, FROM_TYPE, true>(ColumnPtr & column) {  \
+        return column->clone();                                          \
     }
 
```

Comment:
Is it work instead with:
```
    template <bool cast_safe>                                                          \
    ColumnPtr cast_fn<FROM_TYPE, FROM_TYPE, cast_safe>(ColumnPtr & column) {  \
        return column->clone();                                          \
    }
```

Reply:
Have used class's static method instead of this to reduce codes.